app/components/welcome-message.js

```diff
@@ -0,0 +1,26 @@
+import Component from '@ember/component';
+import { inject as service } from '@ember/service';
+import { computed } from '@ember/object';
+import { notEmpty } from '@ember/object/computed';
+
+export default Component.extend({
+  session: service(),
+
+  text: computed('session.currentUser.{email_verified,has_tokens}', function() {
+    const user = this.get('session.currentUser');
+    if (!user || (user.email_verified && user.has_tokens)) return '';
+
+    const textArray = [
+      !user.email_verified && 'verify your email address',
+      !user.email_verified && !user.has_tokens && ' and ',
+      !user.has_tokens && 'create an API token',
+      '!',
+    ].filter(e => !!e);
```

Comment:
the code logic here seems a bit hard to read. do you think it could be simplified in some way?

Reply:
I tried some variants before come to this. For example, a variant with ternary operators looks more ugly:
```
const verifyText = user.email_verified ? '' : 'verify your email address ';
const andText = user.email_verified && user.has_tokens ? '' : 'and ';
const tokenText = user.has_tokens ? '' : 'create an API token';

return `${verifyText}${andText}{$tokenText}`;
```
Moving this calculation logic to template is not right.
I'm open to any ideas on how to improve this code.